### ISSUE TITLE
CacheDecoratorTestCase.test_double_get fails on Python 3.8: 4 != 3

### ISSUE THREAD

**Issue by danc86** [matrix-org/synapse#6200]
### Description

There is a test failure when synapse 1.4.0 is built with Python 3.8.

### Steps to reproduce

- Run the tests using Python 3.8: `PYTHONPATH=. trial-3 tests`

The test suite fails:
```
[...]
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/usr/lib64/python3.8/site-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
    result = g.send(result)
  File "/builddir/build/BUILD/synapse-1.4.0/tests/storage/test__base.py", line 308, in test_double_get
    self.assertEquals(a.func2.cache.cache.pop.call_count, 3)
  File "/usr/lib64/python3.8/site-packages/twisted/trial/_synctest.py", line 432, in assertEqual
    super(_Assertions, self).assertEqual(first, second, msg)
  File "/usr/lib64/python3.8/unittest/case.py", line 912, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/usr/lib64/python3.8/unittest/case.py", line 905, in _baseAssertEqual
    raise self.failureException(msg)
twisted.trial.unittest.FailTest: 4 != 3

tests.storage.test__base.CacheDecoratorTestCase.test_double_get
-------------------------------------------------------------------------------
Ran 780 tests in 194.729s

FAILED (skips=1, failures=1, successes=778)
```

### Version information

- **Version**:
1.4.0

- **Install method**:
running tests from the source tree

- **Platform**:
Fedora rawhide (to become Fedora 32) which includes Python 3.8

**Comment by richvdh** [comment]
We don't currently support python 3.8, so I'm going to go ahead and close this, but if you'd like to submit a PR to fix this, that would be great.

**Comment by hawkowl** [comment]
Reopening as a part of #6223

**Comment by richvdh** [comment]
fixed by #6213
